runtime/manifest.js

```diff
@@ -143,8 +144,17 @@ class Manifest {
     let tags = options.tags || [];
     let subtype = options.subtype || false;
     function typePredicate(view) {
+      let resolvedType = type;
+      let primitiveType = type.isSetView ? type.primitiveType() : type;
+      if (primitiveType.isVariable) {
+        if (!primitiveType.data.isResolved) {
+          return true;
```

Comment:
if type.isVariable & unresolved, then this is the point we debated, we can return true to say any view is potentially valid as far as the type is concerned. otherwise we may need to consider what data is avaiable, eg. [~a] should only match collections

Reply:
updated the code and added test cases where 
[~a] won't accept Thing
and
~a won't accept [Thing]